CRM/Extension/Mapper.php

```diff
@@ -280,7 +280,7 @@ public function getActiveModuleFiles($fresh = FALSE) {
       return [];
     }
 
-    $moduleExtensions = NULL;
+    static $moduleExtensions = NULL;
```

Comment:
Suggestions:

* To ensure that the data is cleared out in-between unit-tests and system-flushes, use `Civi::$statics['foo']` instead of `static $foo`.
* Add a comment to explain why this is used *in addition to* the regular cache. (Whenever I read this line with fresh eyes, I get confused because it mixes a `static` array and a `$cache` object, which would normally be redundant. But in this case it serves a specific purpose.)

I haven't tested, but I think it'd look like this:

```suggestion
    // The list of module files is cached in two tiers; the tiers are slightly different:
    // 1. The persistent tier (`$this->cache->get('...moduleFiles')`) stores names WITHOUT absolute paths.
    // 2. The ephemeral/thread-local tier (`$moduleExtensions`) stores names WITH absolute paths.
    $moduleExtensions = &Civi::$statics[__CLASS__]['moduleExtensions'];
```

Reply:
Thanks @totten I've applied your suggestions. Though I used the static variable a bit differently, after looking at how it was used in `CRM_Case_BAO_Case`.